src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -133,6 +133,13 @@ def __init__(
 
         # override tokenizer to pad to left
         self.tokenizer.padding_side = "left"
+        # read from the tokenizer whether is
+        # uses a prefix to determine the sos token
+        self.sos_token_offset = 0
+        if hasattr(self.tokenizer, "prefix"):
```

Comment:
is this a universal attribute or just for OPT?
also let's move this to a helper function

Reply:
Just for OPT. We do not have this attribute neither for CodeGen nor for BLOOM.